### DISCUSSION TITLE
Running test with npm run command error

### DISCUSSION THREAD

**Discussion by pratikthorve90** [cypress-io/cypress#15883] [Questions and Help] [answered]
I have the below code in my package.json

```
  "description": "this is project 1 for cypress",
  "main": "index.js",
  "scripts": {
    "test": "node_modules/.bin/cypress run",
    "cypress:open": "cypress open"
  },
```

Now when i try to run the same as command like : npm run test / npm run cypress:open i get error , i have attached the logs here.
Can someone please help me in identify what i am doing wrong,

code : https://github.com/pratikthorve90/Cypress

Error as below

`0 verbose cli [
0 verbose cli   'C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\node\\bin\\node.exe',
0 verbose cli   'C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'run',
0 verbose cli   'test'
0 verbose cli ]
1 info using npm@7.6.3
2 info using node@v12.4.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:C:\Users\inpthor\AppData\Roaming\npm\node_modules\npm\npmrc Completed in 2ms
5 timing config:load:builtin Completed in 2ms
6 timing config:load:cli Completed in 2ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1\.npmrc Completed in 1ms
9 timing config:load:project Completed in 2ms
10 timing config:load:file:C:\Users\inpthor\.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:C:\Users\inpthor\AppData\Roaming\npm\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:setUserAgent Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 9ms
19 verbose npm-session 91722ad6a444a633
20 timing npm:load Completed in 21ms
21 timing command:run Completed in 85ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (C:\Users\inpthor\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\promise-spawn\index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:200:13)
22 verbose stack     at maybeClose (internal/child_process.js:1021:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
23 verbose pkgid project1@1.0.0
24 verbose cwd C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1
25 verbose Windows_NT 10.0.17134
26 verbose argv "C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\node\\bin\\node.exe" "C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "test"
27 verbose node v12.4.0
28 verbose npm  v7.6.3
29 error code 1
30 error path C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1
31 error command failed
32 error command C:\WINDOWS\system32\cmd.exe /d /s /c node_modules/.bin/cypress run
33 verbose exit 1
`

**Comment by pratikthorve90** [comment]
Please note i have already tried deleting and reinstalling cypress and also the command works fine if i run it in terminal

**Comment by mccataldo** [comment]
@pratikthorve90 what happens if you execute an npm script that does something other than running or opening cypress?

**Reply by pratikthorve90** [reply]
@mccataldo it worked for the below thought.

` "scripts": {
    "test": "cypress run",
    "cypress:open": "cypress open",
    "say-hello": "echo 'Hello World!'"
  },`

`PS C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1> npm run say-hello

> project1@1.0.0 say-hello
> echo 'Hello World!'

"Hello World!"`

**Comment by bahmutov** [comment, marked as answer]
@pratikthorve90 you have a Windows system according to the NPM stack trace, but the command you posted is has Linux paths

```
"test": "node_modules/.bin/cypress run"
```

In your `package.json` it should simply be:

```
"test": "cypress run"
```

Then `npm test` or `npm t` would work.

**Reply by pratikthorve90** [reply]
@bahmutov i tried the same like below , but still its failing.

```
  "scripts": {
    "test": "cypress run",
    "cypress:open": "cypress open"
  },

```

command used : npm run test.

directory structure where command is run : 

`PS C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1> dir


    Directory: C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----       06-04-2021     07:25                cypress
d-----       08-04-2021     22:57                node_modules
-a----       06-04-2021     07:27            366 cypress.json
-a----       08-04-2021     22:57         228926 package-lock.json
-a----       09-04-2021     09:19            373 package.json
`

error message displayed on cmd prompt 

`PS C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1> npm run test

> project1@1.0.0 test
> cypress run

The system cannot find the path specified.
npm ERR! code 1
npm ERR! path C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c cypress run

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\inpthor\AppData\Local\npm-cache\_logs\2021-04-09T03_49_56_104Z-debug.log`

detailed error as below 

`0 verbose cli [
0 verbose cli   'C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\node\\bin\\node.exe',
0 verbose cli   'C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'run',
0 verbose cli   'test'
0 verbose cli ]
1 info using npm@7.6.3
2 info using node@v12.4.0
3 timing config:load:defaults Completed in 3ms
4 timing config:load:file:C:\Users\inpthor\AppData\Roaming\npm\node_modules\npm\npmrc Completed in 3ms
5 timing config:load:builtin Completed in 3ms
6 timing config:load:cli Completed in 3ms
7 timing config:load:env Completed in 1ms
8 timing config:load:file:C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1\.npmrc Completed in 0ms
9 timing config:load:project Completed in 2ms
10 timing config:load:file:C:\Users\inpthor\.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:C:\Users\inpthor\AppData\Roaming\npm\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 1ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 2ms
18 timing config:load Completed in 16ms
19 verbose npm-session c33de5229e896ad5
20 timing npm:load Completed in 36ms
21 timing command:run Completed in 114ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (C:\Users\inpthor\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\promise-spawn\index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:200:13)
22 verbose stack     at maybeClose (internal/child_process.js:1021:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
23 verbose pkgid project1@1.0.0
24 verbose cwd C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1
25 verbose Windows_NT 10.0.17134
26 verbose argv "C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\node\\bin\\node.exe" "C:\\Users\\inpthor\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "test"
27 verbose node v12.4.0
28 verbose npm  v7.6.3
29 error code 1
30 error path C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1
31 error command failed
32 error command C:\WINDOWS\system32\cmd.exe /d /s /c cypress run
33 verbose exit 1
`

**Reply by pratikthorve90** [reply]
@bahmutov : as suggested above by @mccataldo i tired for a normal npm script and it works for that.

```
  "scripts": {
    "test": "cypress run",
    "cypress:open": "cypress open",
    "say-hello": "echo 'Hello World!'"
  },
```


```
PS C:\Users\inpthor\Desktop\My_Stuff\From_Telus\My_stuff\Cypress\Cypress\Code\Project1> npm run say-hello

> project1@1.0.0 say-hello
> echo 'Hello World!'

"Hello World!"
```
